### PR TITLE
chore: update BBN version to FP addr is the signer

### PR DESCRIPTION
**Pull request by RafilxTenfen**
- The key for Babylon in the config needs to be the FP key to be the signer
- E2E testing now if the FP needs to sign something it needs to update the ClientController to be able to sign msgs with the corresponding key

Related: https://github.com/babylonchain/babylon/pull/678

### CODE REVIEW

**Review by SebastianElvis — APPROVED** [review]

File: clientcontroller/interface.go [hunk 1]
```diff
@@ -19,11 +19,10 @@ const (
 )
 
 type ClientController interface {
-
 	// RegisterFinalityProvider registers a finality provider to the consumer chain
 	// it returns tx hash and error
 	RegisterFinalityProvider(
-		chainPk []byte,
+		// the address of the finality provider will be the signer of the msg.
```
Comment (gitferry) [inline clientcontroller/interface.go:25]: seems this comment can be put to the doc string? Seems strange to be close to `fpPk`.

File: keyring/keyringcontroller.go [hunk 2]
```diff
@@ -121,13 +122,19 @@ func (kc *ChainKeyringController) CreateChainKey(passphrase, hdPath, mnemonic st
 // CreatePop creates proof-of-possession of Babylon and BTC public keys
 // the input is the bytes of BTC public key used to sign
 // this requires both keys created beforehand
-func (kc *ChainKeyringController) CreatePop(btcPrivKey *btcec.PrivateKey, passphrase string) (*bstypes.ProofOfPossession, error) {
-	bbnPrivKey, err := kc.GetChainPrivKey(passphrase)
+func (kc *ChainKeyringController) CreatePop(fpAddr sdk.AccAddress, btcPrivKey *btcec.PrivateKey) (*bstypes.ProofOfPossessionBTC, error) {
+	return bstypes.NewPoPBTC(fpAddr, btcPrivKey)
+}
+
+// Address returns the address inside the config.
```
Comment (gitferry) [inline keyring/keyringcontroller.go:129]:
```suggestion
// Address returns the address from the keyring
```

**Review by gitferry — COMMENTED** [review]
Code-wise looks good to me! Maybe we can also have `docs/finality-provider.md` updated accordingly in this PR?

File: clientcontroller/interface.go (hunk 1 above)
Comment (RafilxTenfen) [reply clientcontroller/interface.go]: https://github.com/babylonchain/finality-provider/pull/435/commits/2d56729d586b1939a6942f5dfe2d60cb84713d0d

File: keyring/keyringcontroller.go (hunk 2 above)
Comment (RafilxTenfen) [reply keyring/keyringcontroller.go]: https://github.com/babylonchain/finality-provider/pull/435/commits/2d56729d586b1939a6942f5dfe2d60cb84713d0d

File: docs/finality-provider.md [hunk 3]
```diff
@@ -52,10 +52,12 @@ Below are some important parameters of the `fpd.conf` file.
 
 **Note**:
 The configuration below requires pointing to the path where this keyring is
-stored `KeyDirectory`. This `Key` field stores the key name used for interacting with
-the consumer chain and will be specified along with the
-`KeyringBackend` field in the next [step](#3-add-key-for-the-consumer-chain). So we
-can ignore the setting of the two fields in this step.
+stored `KeyDirectory`. This `Key` field stores the key name of the finality
+provider wallet that is going to be used for interacting with the consumer
+chain and to sign the messages of your finality provider, it will be specified
```
Comment (gitferry) [inline docs/finality-provider.md:57]:
```suggestion
chain and to sign the messages of your finality provider. It will be specified
```

**Review by gitferry — APPROVED** [review]
Thanks for the changes. LGTM!

**Pull request merged**